flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartitionReadSchedulerTest.java

```diff
@@ -192,6 +194,53 @@ public void testOnReadBufferRequestError() throws Exception {
         assertAllResourcesReleased();
     }
 
+    @Test
+    public void testIncreaseDeadlineWhenRecycleBuffer() throws Exception {
```

Comment:
About the test, I think we can test at least 3 cases:
1. buffer request timeout without buffer recycle.
2. buffer request timeout refreshed on buffer recycle.
3. buffer request timeout after buffer recycle.

Reply:
Ok, I have added the above tests. For the third case, I added 2 tests.